hazelcast/src/main/java/com/hazelcast/map/impl/MapPartitionAwareService.java

```diff
@@ -36,26 +36,31 @@ class MapPartitionAwareService implements PartitionAwareService {
 
     private final MapServiceContext mapServiceContext;
     private final NodeEngine nodeEngine;
-    private final ProxyService proxyService;
+    private final EventServiceImpl eventService;
+    private final ILogger logger;
 
     MapPartitionAwareService(MapServiceContext mapServiceContext) {
         this.mapServiceContext = mapServiceContext;
         this.nodeEngine = mapServiceContext.getNodeEngine();
-        this.proxyService = this.nodeEngine.getProxyService();
+        this.eventService = (EventServiceImpl) this.nodeEngine.getEventService();
+        this.logger = nodeEngine.getLogger(MapPartitionAwareService.class.getName());
     }
 
     @Override
     public void onPartitionLost(IPartitionLostEvent partitionLostEvent) {
         final Address thisAddress = nodeEngine.getThisAddress();
         final int partitionId = partitionLostEvent.getPartitionId();
 
-        Collection<DistributedObject> result = proxyService.getDistributedObjects(MapService.SERVICE_NAME);
-
-        for (DistributedObject object : result) {
-            final MapProxyImpl mapProxy = (MapProxyImpl) object;
-            final String mapName = mapProxy.getName();
+        EventServiceSegment eventServiceSegment = eventService.getSegment(MapService.SERVICE_NAME, false);
+        if (eventServiceSegment == null) {
+            logger.warning("No service registration found for " + MapService.SERVICE_NAME);
```

Comment:
Do we need this log? I assume this code will be run no matter whether a partition lost listener is added or not. (The onPartitionLost method, through the InternalPartitionLostEventPublisher). And, if the user hasn't added any listeners to any of the maps yet, or not using the map at all we will hit this line. 

I believe it can confuse the user. I am in favor of removing it, what do you think?

Reply:
Yes, you are right. Makes sense!
Thanks!